lib/child_process.js

```diff
@@ -35,9 +35,9 @@ exports.fork = function(modulePath /*, args, options*/) {
   if (pos < arguments.length && arguments[pos] != null) {
     if (typeof arguments[pos] !== 'object') {
       throw new TypeError('Incorrect value of args option');
-    } else {
-      options = util._extend({}, arguments[pos++]);
     }
+
+    options = util._extend({}, arguments[pos++]);
```

Comment:
This change means we access an out-of-bounds `pos` on `arguments`, which recent PRs have claimed is a deoptimization. Any concern with that?